src/main/java/org/elasticsearch/index/query/QueryFilterParser.java

```diff
@@ -19,17 +19,14 @@
 
 package org.elasticsearch.index.query;
 
-import org.apache.lucene.search.Filter;
+import org.apache.lucene.search.ConstantScoreQuery;
 import org.apache.lucene.search.Query;
-import org.apache.lucene.search.QueryWrapperFilter;
 import org.elasticsearch.common.inject.Inject;
 
 import java.io.IOException;
 
-/**
- *
- */
-public class QueryFilterParser implements FilterParser {
+@Deprecated
+public class QueryFilterParser implements QueryParser {
```

Comment:
The reasoning is that Java users will have a nice compile error so they can fix their code quite easily. On the other hand consumers of the REST layer can't so I'm keeping it for backward compatibility. But you made me notice I forgot to talk about it in the migration notes, I'll fix.